File: 백준/Silver/22233. 가희와 키워드/가희와 키워드.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());

        // 🔹 HashSet 대신 LinkedHashSet 사용 (입력 순서 유지 가능)
        Set<String> keywords = new LinkedHashSet<>();
        for (int i = 0; i < N; i++) {
            keywords.add(br.readLine());
        }

        for (int i = 0; i < M; i++) {
            String[] words = br.readLine().split(",");
            for (String word : words) {
                keywords.remove(word); // ✅ 평균 O(1)로 최적화됨
            }
            bw.write(keywords.size() + "\n");
        }

        bw.flush();
        bw.close();
        br.close();
    }
}

File: 백준/Silver/22233. 가희와 키워드/가희와 키워드.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];
rl.on("line", (line) => {
  input.push(line);
}).on("close", () => {
  let [N, M] = input[0].split(" ").map(Number);

  // 🔹 Java의 LinkedHashSet과 유사한 Set 사용 (삽입 순서 유지)
  let keywords = new Set();

  for (let i = 1; i <= N; i++) {
    keywords.add(input[i]);
  }

  let output = [];
  for (let i = N + 1; i <= N + M; i++) {
    let words = input[i].split(",");
    for (let word of words) {
      keywords.delete(word); // ✅ 평균 O(1)으로 삭제
    }
    output.push(keywords.size);
  }

  console.log(output.join("\n"));
  process.exit();
});
